recipes-caddy/caddy/github.com-burntsushi-toml.bb

```diff
@@ -4,9 +4,8 @@ GO_IMPORT = "github.com/BurntSushi/toml"
 
 inherit go
 
-SRC_URI = "git://github.com/BurntSushi/toml;protocol=https;destsuffix=${PN}-${PV}/src/${GO_IMPORT}"
+SRC_URI = "git://github.com/BurntSushi/toml;protocol=https"
 SRCREV = "${AUTOREV}"
 LICENSE = "DWTFYW"
-LIC_FILES_CHKSUM = "file://src/${GO_IMPORT}/COPYING;md5=389a9e29629d1f05e115f8f05c283df5"
-
-FILES_${PN} += "${GOBIN_FINAL}/*"
+LIC_FILES_CHKSUM = "file://COPYING;md5=389a9e29629d1f05e115f8f05c283df5"
+S = "${WORKDIR}/git"
```

Comment:
Maybe this could be simplified even futher similar to how the [`pypi.bbclass`](https://github.com/openembedded/meta-openembedded/blob/master/meta-python/classes/pypi.bbclass) is implemented.
 
For example in a `gopkg.bbclass` it could:
- the `SRC_URI` could be derived from the import path
- `S` could be set by default
- fill the `HOMEPAGE` variable

There would only be a minimal set of information required for a standard conform packages, like it is for python for example: https://github.com/openembedded/meta-openembedded/blob/master/meta-python/recipes-devtools/python/python-chardet.inc

Example:
```
GO_IMPORT = "github.com/BurntSushi/toml"
LICENSE = "DWTFYW"
LIC_FILES_CHKSUM = "file://COPYING;md5=389a9e29629d1f05e115f8f05c283df5"

SRCREV = "${AUTOREV}"

inherit gopkg
```